slang.h

```diff
@@ -3857,6 +3857,9 @@ namespace slang
         PreprocessorMacroDesc const*    preprocessorMacros = nullptr;
         SlangInt                        preprocessorMacroCount = 0;
 
+            /** The line directive mode for output source code.
+            */
+        SlangLineDirectiveMode lineDirectiveMode = SLANG_LINE_DIRECTIVE_MODE_DEFAULT;
```

Comment:
This realistically belongs in `TargetDesc`.

Also note that adding fields to these `Desc` types creates a binary backwards-compatibility issue, and it isn't actually something we've done before. In principle, this change would mean we can't drop a new `slang.dll` into an application compiled against an older version of `slang.h`.

I'm tentatively okay with letting this slip through for now just to see what happens, but we really need to put a true binary-compatibility plan into place for this stuff.